### PR TITLE
tests, network, probes: Use unprivileged pod

### PR DESCRIPTION
**Pull request by nirdothan**
There is no reason to use a privileged pod with HostPid , root user and a fully permissive security context, in order to serve as a probe for a VM. Use  unprivileged regular pod instead.

### What this PR does
Before this PR:
NONE
After this PR:

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

**Review by akalenyu — COMMENTED** [review]
So if you drop the manual cleanup https://github.com/kubevirt/kubevirt/blob/4709e1116dc173bcd51688aa1f5df0cc0761afd7/tests/network/probes.go#L89-L91 the pods will actually stick around for you to debug them. You don't have to manually clean them up since we have a global helper cleaning up the namespaces https://github.com/kubevirt/kubevirt/blob/4709e1116dc173bcd51688aa1f5df0cc0761afd7/tests/testsuite/namespace.go#L153-L158

**Comment by akalenyu** [comment]
> /uncc @EdDev /uncc @RamLavi /uncc @akalenyu Moving to to draft, trying to get over the flakes lane issue

Long shot but maybe the flake lane is missing some component which is available in the sig-network lane
Otherwise, it could also just be flaking

**Review by RamLavi — COMMENTED** [review]
in `tests, network, probes: Bug fix: bad use of global pointer` commit, please consider initializing the `HTTPGet`, `TCPSocket` on the global initializer function `createTCPProbe` , instead of "undoing" something a test formerly set. 
Doing so will make things cleared and more stateless.

**Comment by ormergi** [comment]
@nirdothan please consider posting dedicated PR for the second commit, it doesnt seem to be related to getting rid of host-network/privileged pod usage. 
Please note any discussion on second commit could stall the original intention of this PR.

File: tests/network/probes.go [hunk 1]
```diff
@@ -73,7 +73,7 @@ var _ = SIGDescribe("[ref_id:1182]Probes", func() {
 		const (
 			period         = 5
 			initialSeconds = 5
-			timeoutSeconds = 1
+			timeoutSeconds = 5
```
Comment (ormergi) [inline tests/network/probes.go:76]: Does changing timeouts is intentional or leftover from troubleshooting?

File: tests/network/probes.go [hunk 2]
```diff
@@ -193,7 +195,7 @@ var _ = SIGDescribe("[ref_id:1182]Probes", func() {
 				vmi = createReadyAlpineVMIWithLivenessProbe(livenessProbe)
 			} else if !isExecProbe(livenessProbe) {
 				By(specifyingVMLivenessProbe)
-				vmi = createReadyAlpineVMIWithLivenessProbe(livenessProbe)
+				vmi = createReadyAlpineVMIWithLivenessProbe(undoPointIpv6ProbeToSupportPod(livenessProbe))
```
Comment (ormergi) [inline tests/network/probes.go:198]: Despite making the tests hard to follow and somewhat unpredictable, does the fact Prob{} instances are being reused in the test introduces some bug?

**Review by ormergi — APPROVED** [review]
Thanks for the PR, LGTM

**Comment by ormergi** [comment]
@nirdothan Please updated the PR description following recent changes.

**Review by EdDev — COMMENTED** [review]
Thank you!

/approve
